### ISSUE TITLE
`ConsMDP` will become just a data structure (refactor solvers` interface)

### ISSUE THREAD

**Issue by xblahoud** [xblahoud/FiMDP#22]
The solvers should be more friendly to use from CMDP objects and should return strategies by default.

**Comment by xblahoud** [comment]
Currently, `ConsMDP` class implements `get_positiveReachability`, `get_buchi` and similar functions to compute the minimal levels needed to satisfy the specified objective. It also has the `energy_levels` attribute that stores a pointer to the last solver used by invoking `get_buchi` or some similar function. The purpose of this attribute was for visualization of the computed values.

We will remove all of that. With that, also the `def_EL_class` will be removed as it will be no longer needed.

In the next release, the ConsMDP will be used just as a data structure.

The new interface will require solvers to be used explicitely. One of the reasons is that we now have multiple of them and they need different parameters so a unified interface through `ConsMDP` will not be transparent anyway.

An example of the intended use follows. Note that now we will allow to visualize solvers using `show()` function (or by `__repr__()` in notebooks). This will visualize the underlying mdp with the highlighted targets and (if already computed) the requested values.

```python
from fimdp.core import ConsMDP
from fimdp.objectives import BUCHI
from fimdp.solvers import BasicES

mdp = ConsMDP(...)
# ... build the mdp

# Compute action selector and visualize the minimal levels
solver = BasicES(mdp, cap=40, targets=...)
selector = solver.get_selector_for(BUCHI)
solver.show()
```

**Comment by xblahoud** [comment]
Solved by 08ae83d.
